### PR TITLE
New warning & negative background colors

### PR DESCRIPTION
**Pull request by FlyersPh9**
## Changes

- New `$bg-*-1` and `$bg-*-2` colors for warning (orange) & negative (red).
  - This was done to make the colors more consistent in their vibrancy.  This was done using `oklch` in #2053 but converted back to `hsl` for this PR.
  - Informational (blue) & positive (green) were not changed because they are frequently used for button background colors and the change would have been so noticeable warranting a major release.
  - `$bg-*-0` was not changed for warning (orange) & negative (red) because it would have stuck out compared to the unchanged informational (blue) & positive (green).

## Testing

- Ran visual tests on CSS and React.
- Used Chrome dev tools to ensure passing WCAG AA & AAA.

## Docs

n/a

### CODE REVIEW

**Review by Ben-Pusey-Bentley — APPROVED** [review]

File: .changeset/funny-colts-cheer.md
Comment (mayank99) [inline .changeset/funny-colts-cheer.md:1]: The actual variable names should also be listed in this changeset.

File: .changeset/thin-eggs-dream.md
Comment (mayank99) [inline .changeset/thin-eggs-dream.md:1]: CSS changeset not needed.

File: packages/itwinui-variables/src/themes/dark-hc.scss [hunk 1]
```diff
@@ -34,12 +34,12 @@ $bg-positive-1: hsl(118 81% 21%);
 $bg-positive-2: hsl(118 76% 28%);
 
 $bg-warning-0: hsl(60 1% 26%);
-$bg-warning-1: hsl(33 94% 25%);
-$bg-warning-2: hsl(33 90% 33%);
+$bg-warning-1: hsl(41.86 100% 21%);
```
Comment (mayank99) [inline packages/itwinui-variables/src/themes/dark-hc.scss:37]: Should we round these hue values?

(images will likely change as a result)

**Review by mayank99 — APPROVED** [review]
Some minor comments, but LGTM otherwise!

File: .changeset/funny-colts-cheer.md
Comment (FlyersPh9) [reply .changeset/funny-colts-cheer.md:1]: Listed the variable names effected.

File: .changeset/thin-eggs-dream.md
Comment (FlyersPh9) [reply .changeset/thin-eggs-dream.md:1]: Deleted CSS changeset.

File: packages/itwinui-variables/src/themes/dark-hc.scss (hunk 1 above)
Comment (FlyersPh9) [reply packages/itwinui-variables/src/themes/dark-hc.scss]: Rounded the values… but surprisingly image tests didn't change?

**Review by rohan-kadkol — APPROVED** [review]
Reconfirmed that the conversions from `oklch` to `hsl` were correct. The new colors LGTM!  🚀

**Pull request merged**